ca/server.go

```diff
@@ -202,6 +202,9 @@ func (s *Server) IssueNodeCertificate(ctx context.Context, request *api.IssueNod
 					Membership: api.NodeMembershipAccepted,
 				},
 			}
+			if request.Spec != nil {
+				node.Spec.Availability = request.Spec.Availability
```

Comment:
We can't let the node requesting a certificate decide its own role. That has to be determined from the join token. This is why it's very important that we not honor all the fields.